internal/db/event_logs.go

```diff
@@ -679,35 +695,48 @@ FROM (
     name,
     -- Postgres 9.6 needs to go from text to integer (i.e. can't go directly to integer)
     (argument->'durationMs')::text::integer as latency,
-    CASE WHEN user_id = 0
-      -- It's faster to group by an int rather than text, so we convert
-      -- the anonymous_user_id to an int, rather than the user_id to text.
-      THEN ('x'||substr(md5(anonymous_user_id), 1, 8))::bit(32)::int
-      ELSE user_id
-    END AS user_id,
-    DATE_TRUNC('month', TIMEZONE('UTC', timestamp)) as month,
-    DATE_TRUNC('week', TIMEZONE('UTC', timestamp) + '1 day'::interval) - '1 day'::interval as week,
-    DATE_TRUNC('day', TIMEZONE('UTC', timestamp)) as day,
-    DATE_TRUNC('month', TIMEZONE('UTC', %s::timestamp)) as current_month,
-    DATE_TRUNC('week', TIMEZONE('UTC', %s::timestamp) + '1 day'::interval) - '1 day'::interval as current_week,
-    DATE_TRUNC('day', TIMEZONE('UTC', %s::timestamp)) as current_day
+    ` + aggregatedUserIDQueryFragment + ` AS user_id,
+    ` + makeDateTruncExpression("month", "timestamp") + ` as month,
+    ` + makeDateTruncExpression("week", "timestamp") + ` as week,
+    ` + makeDateTruncExpression("day", "timestamp") + ` as day,
+    ` + makeDateTruncExpression("month", "%s::timestamp") + ` as current_month,
+    ` + makeDateTruncExpression("week", "%s::timestamp") + ` as current_week,
+    ` + makeDateTruncExpression("day", "%s::timestamp") + ` as current_day
   FROM event_logs
-  WHERE timestamp >= DATE_TRUNC('month', TIMEZONE('UTC', %s::timestamp)) AND name IN (
-    'codeintel.lsifHover',
-    'codeintel.searchHover',
-    'codeintel.lsifDefinitions',
-    'codeintel.searchDefinitions',
-    'codeintel.lsifReferences',
-    'codeintel.searchReferences',
-    'search.latencies.literal',
-    'search.latencies.regexp',
-    'search.latencies.structural',
-    'search.latencies.file',
-    'search.latencies.repo',
-    'search.latencies.diff',
-    'search.latencies.commit',
-    'search.latencies.symbol'
-  )
+  WHERE
+    timestamp >= ` + makeDateTruncExpression("month", "%s::timestamp") + `
+	AND name IN (` + strings.Join(append(append([]string(nil), codeIntelEventNames...), searchEventNames...), ", ") + `)
 ) events
 GROUP BY name, current_month, current_week, current_day
 `
+
+// userIDQueryFragment is a query fragment that can be used to return the anonymous user id.
+// If no anonymous user id is set then the user id is returned.
```

Comment:
Hmm, not 100% accurate I think.
```suggestion
// userIDQueryFragment is a query fragment that can be used to return the anonymous user ID
// when the user ID is not set (i.e. 0).
```